protobuf/src/main/java/io/grpc/protobuf/ProtoUtils.java

```diff
@@ -73,7 +74,12 @@ public T parse(InputStream stream) {
           }
         }
         try {
-          return parser.parseFrom(stream);
+          // Pre-create the CodedInputStream so that we can remove the size limit restriction
+          // when parsing.
+          CodedInputStream codedInput = CodedInputStream.newInstance(stream);
+          codedInput.setSizeLimit(Integer.MAX_VALUE);
+
+          return parser.parseFrom(codedInput);
```

Comment:
In the documentation for `parseFrom(CodedInputStream)`:
>    Note:  The caller should call CodedInputStream#checkLastTagWas(int) after calling this to verify that the last tag seen was the appropriate end-group tag, or zero for EOF.

In my comment in #832, I say:
> The only gotcha I see is that we need to call [`checkLastTagWas(0)`](https://github.com/google/protobuf/blob/master/java/src/main/java/com/google/protobuf/AbstractParser.java#L194) afterward.